api/src/opentrons/__init__.py

```
@@ -159,14 +159,22 @@ async def _blink() -> None:
             await hardware.set_lights(button=False)
             await asyncio.sleep(0.5)
 
+    # While the hardware was initializing in _create_hardware_api(), it blinked the
+    # front button light. But that blinking stops when the completed hardware object
+    # is returned. Do our own blinking here to keep it going while we home the robot.
     blink_task = asyncio.create_task(_blink())
 
-    if not ff.disable_home_on_boot():
-        log.info("Homing Z axes")
-        await hardware.home_z()
-
-    blink_task.cancel()
-    await asyncio.gather(blink_task, return_exceptions=True)
-    await hardware.set_lights(button=True)
-
-    return hardware
+    try:
+        if not ff.disable_home_on_boot():
+            log.info("Homing Z axes")
+            await hardware.home_z()
+
+        await hardware.set_lights(button=True)
+
+        return hardware
+    finally:
+        blink_task.cancel()
+        try:
+            await blink_task
+        except asyncio.CancelledError:
+            pass
```

Two changes here:

1. Don't accidentally leak the background `blink_task` if this hardware code raises an exception.
2. Avoid swallowing *all* exceptions from `blink_task`; use `await blink_task` instead of `asyncio.gather(blink_task, return_exceptions=True)`.

See also the other thread about the `await blink_task` caveat.